gst/datarepo/gstdatareposrc.c

```diff
@@ -332,6 +348,60 @@ gst_data_repo_src_set_file_path (GstDataRepoSrc * src, const int prop,
   }
 }
 
+/**
+ * @brief Function to set tensors sequence
+ */
+static gboolean
+gst_data_repo_src_set_tensors_sequence (GstDataRepoSrc * src)
+{
+  gchar **strv = NULL;
+  guint length;
+  guint i = 0;
+
+  g_return_val_if_fail (src != NULL, FALSE);
+  g_return_val_if_fail (src->tensors_seq_str != NULL, FALSE);
+
+  /* not use NNS_TENSOR_SIZE_LIMIT */
+  strv = g_strsplit (src->tensors_seq_str, ",", -1);
+
+  length = g_strv_length (strv);
+  if (length > NNS_TENSOR_SIZE_LIMIT) {
+    GST_ERROR_OBJECT (src, "The total number of indices exceeded %d.",
+        NNS_TENSOR_SIZE_LIMIT);
+    goto error;
+  }
+
+  while (strv[i] != NULL && strlen (strv[i]) > 0) {
+    src->tensors_seq[i] = atoi (strv[i]);
```

Comment:
How about use sth like `(guint) g_ascii_strtoll (strv[i], NULL, 10)` rather than vulnerable `atoi`?

Reply:
okay, I'll try and test